scripts/zones/Lower_Jeuno/npcs/Ghebi_Damomohe.lua

```diff
@@ -6,91 +6,81 @@
 -- @zone 245
 -- !pos 16 0 -5
 -----------------------------------
-package.loaded["scripts/zones/Lower_Jeuno/TextIDs"] = nil;
-package.loaded["scripts/globals/settings"] = nil;
------------------------------------
-require("scripts/globals/settings");
-require("scripts/globals/shop");
-require("scripts/globals/titles");
-require("scripts/globals/keyitems");
-require("scripts/globals/quests");
-require("scripts/zones/Lower_Jeuno/TextIDs");
+package.loaded["scripts/zones/Lower_Jeuno/TextIDs"] = nil
+package.loaded["scripts/globals/settings"] = nil
 -----------------------------------
+require("scripts/zones/Lower_Jeuno/TextIDs")
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+require("scripts/globals/npc_util")
+require("scripts/globals/titles")
+require("scripts/globals/quests")
+require("scripts/globals/shop")
 
 function onTrade(player,npc,trade)
-    if (player:getQuestStatus(JEUNO,TENSHODO_MEMBERSHIP)~= QUEST_COMPLETED and trade:hasItemQty(548,1) == true and trade:getItemCount() == 1) then
-        player:startEvent(108); -- Finish Quest (don't need fame or starting quest)
-    elseif (player:hasKeyItem(dsp.ki.PSOXJA_PASS)==false and (trade:hasItemQty(1692,1) or trade:hasItemQty(1694,1) or trade:hasItemQty(1693,1)) and trade:getItemCount() == 1) then
-        --Carmine Chip (Ex) - Snow Lizard or Frost Lizard
-            --Gray Chip (Ex) - Diremite Stalker (at Tower near Ranguemont), Diremite Assaulter, Diremite
-            --Cyan Chip (Ex) - Treasure Chest Mimics
-            player:startEvent(52);
+    if player:getQuestStatus(JEUNO,TENSHODO_MEMBERSHIP) ~= QUEST_COMPLETED and npcUtil.tradeHas(trade, 548) then
+        -- Finish Quest: Tenshodo Membership (Invitation)
+        player:startEvent(108)
+    elseif player:getCurrentMission(COP) == DARKNESS_NAMED and
+           not player:hasKeyItem(dsp.ki.PSOXJA_PASS) and
+           player:getVar("PXPassGetGems") == 1 and
+           npcUtil.tradeHas(trade, 1692) or npcUtil.tradeHas(trade, 1694) or npcUtil.tradeHas(trade, 1693) then
```

Comment:
I think AND is processed before OR in order of operations, so you'd want to put parentheses around this group of three tradeHas's

Reply:
Ah you're right, good catch.